src/main/java/com/sap/oss/phosphor/fosstars/data/github/SecuregoDataProvider.java

```diff
@@ -0,0 +1,263 @@
+
+    // ideally, we're looking for a GitHub action that runs Securego scan on pull requests
+    // but if we just find an action that runs Securego scans with or without rules is also fine
+    for (Path configPath : findGitHubActionsIn(repository)) {
+      try (InputStream content = Files.newInputStream(configPath)) {
+        Map<String, Object> githubAction = Yaml.readMap(content);
+        if (triggersScan(githubAction)) {
+          runsSecurego = RUNS_SECUREGO_SCANS.value(true);
+          if (runsOnPullRequests(githubAction)) {
+            usesSecuregoScanChecks = USES_SECUREGO_SCAN_CHECKS.value(true);
+          }
+          if (runsWithSelectedRules(githubAction)) {
+            usesSecuregoWithSelectedRules = USES_SECUREGO_WITH_RULES.value(true);
+            break;
+          }
+        }
+      }
+    }
+
+    return ValueHashSet.from(runsSecurego, usesSecuregoWithSelectedRules, usesSecuregoScanChecks);
+  }
+
+  /**
+   * Checks if the Github action uses Securego sacn with rules.
+   *
+   * @param githubAction A config of the action
+   * @return True if the Github Action has config to run Securego with rules.
+   */
+  private boolean runsWithSelectedRules(Map<?, ?> githubAction) {
+    return Optional.ofNullable(githubAction.get("jobs"))
+        .filter(Map.class::isInstance)
+        .map(Map.class::cast)
+        .map(jobs -> jobs.values())
+        .filter(Iterable.class::isInstance)
+        .map(Iterable.class::cast)
+        .map(SecuregoDataProvider::checkAllJobs)
+        .get();
+  }
+
+  /**
+   * Checks if any step in a collection of jobs triggers a Securego scan with rules.
+   *
+   * @param jobs The collection of jobs from GitHub action.
+   * @return True if a step triggers a Securego scan with rules, false otherwise.
+   */
+  private static boolean checkAllJobs(Iterable<?> jobs) {
+    return IteratorUtils.toList(jobs.iterator()).stream()
+        .filter(Map.class::isInstance)
+        .map(Map.class::cast)
+        .map(job -> job.get("steps"))
+        .filter(Iterable.class::isInstance)
+        .map(Iterable.class::cast)
+        .map(SecuregoDataProvider::checkAllSteps)
+        .findAny()
+        .get();
+  }
+
+  /**
+   * Checks if a collection of steps from a GitHub action contains a step that triggers a Securego
+   * scan with rules.
+   *
+   * @param steps The steps to be checked.
+   * @return True if the steps contain a step that triggers a Securego scan with rules, false
+   *     otherwise.
+   */
+  private static boolean checkAllSteps(Iterable<?> steps) {
+    AtomicBoolean usesWithRules = new AtomicBoolean(false);
+    IteratorUtils.toList(steps.iterator()).stream()
+        .filter(Map.class::isInstance)
+        .map(Map.class::cast)
+        .forEach(
+            step -> {
+              if (securegoConfigExists(step.get("uses"))) {
+                usesWithRules.set(hasArgsWithRules(step.get("with")));
```

Comment:
how do you know `step.get("with")` is present within `uses`

Reply:
I see that it is being checked with `Optional.ofNullable`

Reply:
See line 139, I am checking if "uses" present in the step and then only I am checking "with" keys value

Reply:
I was asking how can you say that "with" keys value is present if "uses" key is present?

Reply:
I am not sure if this is present so used this as nullable object in the calling function

Reply:
Added test https://github.com/SAP/fosstars-rating-core/pull/863/files#diff-387aa3ffb31c81a8a078c13084ac4e10e83e3afcae79e2857bc5b7d3238e30c0R146